### PR TITLE
Cleanup prototype datasets CI and related things

### PR DESCRIPTION
**Pull request by pmeier**
As sad as it is, this PR does three things:

1. Remove the datasets v2 tests from CI
2. Remove the decoding transforms since encoded data can only come from datasets v2
3. Move the `features.Encoded*` into the `torchvision.prototype.datasets` namespace, since they are only used there.

cc @seemethere @bjuncek

### CODE REVIEW

File: .circleci/config.yml.in [hunk 1]
```diff
@@ -152,15 +152,6 @@ commands:
           args: --no-build-isolation <<# parameters.editable >> --editable <</ parameters.editable >> .
           descr: Install torchvision <<# parameters.editable >> in editable mode <</ parameters.editable >>
 
-  install_prototype_dependencies:
```
Comment (NicolasHug) [inline .circleci/config.yml.in:155]: thanks Philip, Do you mind just leaving that one last time, to make sure CI is green (and then remove it)

**Review by datumbox — APPROVED** [review]
The changes on the Transforms LGTM, thanks.

File: .circleci/config.yml.in (hunk 1 above)
Comment (pmeier) [reply .circleci/config.yml.in:155]: You commented on the CircleCi config, but these test continue to run. We only removed the dependencies there. You meant that I should run the datasets tests again, right?

Comment (pmeier) [reply .circleci/config.yml.in:155]: I've temporarily reinstated the prototype datasets CI in 215fb185cf6be5be7adf0388116c77acc9a5d3f3. LMK if that is not what you wanted.

Comment (pmeier) [reply .circleci/config.yml.in:155]: CI is red, but for an unrelated failure on the transforms tests. Datasets tests are [green](https://github.com/pytorch/vision/actions/runs/3443182196/jobs/5744471832#step:11:110).

**Review by NicolasHug — APPROVED** [review]
Thanks Philip

I see some prototype failures in the `[REVERT ME]` diff that aren't present in the latest one. I assume they were fixed in-between in https://github.com/pytorch/vision/pull/6934 ? If that's the case, LGTM. Thanks!

**Pull request merged**